packages/docusaurus/src/commands/swizzle/components.ts

```diff
@@ -38,6 +38,30 @@ export type ThemeComponents = {
 const formatComponentName = (componentName: string): string =>
   componentName.replace(/[/\\]index\.[jt]sx?/, '').replace(/\.[jt]sx?/, '');
 
+function sortComponentNames(componentNames: string[]): string[] {
+  return componentNames.sort(); // Algo may change?
+}
+
+// Even if a folder is not directly a component,
+// we still want to be able to swizzle parent folders of any component
+// See https://github.com/facebook/docusaurus/pull/7175#issuecomment-1103757218
+function getMissingIntermediateComponentFolderNames(
```

Comment:
Let's start getting in the habit of using JSDoc if we are going to comment functions/variables, even for internal functions. This makes the DX better.

Reply:
added some

not sure what's the benefits for this in terms of DX, with my IDE I don't really see any improvement

Reply:
When you hover over the call to `getMissingIntermediateComponentFolderNames`, you see its description. You don't necessarily need the `@param` tag though, just turn `//` into `/**`

Reply:
👍 fine for using `/**`

My IDE doesn't show description on command+hover (but this surprises me, I'm pretty sure it many years ago when I was a Java dev 🤷‍♂️ )